### PR TITLE
Redesigned JFMessageDialog

### LINKED ISSUE #133: Watch from/Start over dialog sometimes automatically continues after clicking an episode
**Describe the bug**
I can see the new box pop up to select if I want to continue watching from where I was or start over, but most of the time it is auto selecting

**To Reproduce**
1. Watch a few minutes of an episode
2. Go back to the Season list
3. Reselect the same episode
4. Get automatically moved into the episode at last watched time

**Expected behavior**
Dialog to take input

**Logs**
I'm not getting anything from the telnet

For context this is a 40~ minute tv show and I was stopping and going back in about 4 minutes in

### PR DESCRIPTION
**Pull request by thomabx**
Redesigned JFMessageDialog to allow custom theming and have more consistent appearance with the Web UI. Dialog should automatically stretch to message length. 

This should fix #133.

**Changes**
Pressing back on the resume playback dialog supports backpress. This is required by #122 §4.7
Call to CreateVideoGroup in main.brs waits for valid video before removing current focus.
Removed redundant code form the two current dialog functions into one.
Added 9-patch rounded image with 10px corners

### CODE REVIEW

**Review by cewert — CHANGES_REQUESTED** [review]
I think these look great and it looks like you cleaned up the code a lot. Nice job!

`message_dialog()` worked great

`option_dialog()` seemed to have trouble with an episode. I chose to start it from the beginning but it acted like I hit the back button. Resuming worked fine. Maybe it's an episode thing? I'm drawing a blank for something else to test with more than one option

**Comment by cewert** [comment]
There is a typo in the resume episode prompt - line 122 of `VideoPlayer.brs` should be beginning not begining

File: source/VideoPlayer.brs [hunk 1]
```diff
@@ -118,7 +119,7 @@ function getCaptionMode() as string
 end function
 
 'Opens dialog asking user if they want to resume video or start playback over
-function startPlayBackOver(time as LongInteger) as boolean
+function startPlayBackOver(time as LongInteger) as integer
```
Comment (thomabx) [inline source/VideoPlayer.brs:122]: This should fix selecting to start a video over.

Comment (cewert) [reply source/VideoPlayer.brs:122]: Yep this fixed it :+1:

**Review by cewert — APPROVED** [review]

File: source/utils/misc.brs [hunk 2]
```diff
@@ -57,26 +57,65 @@ function div_ceiling(a as integer, b as integer) as integer
...
+    dialog.message = message
+  end if
 
-  m.scene.dialog = dialog
-  m.scene.dialog.setFocus(true)
+  dialog.visible = true
+  dialog.setFocus(true)
+  m.scene.appendChild(dialog)
+
+  port = CreateObject("roMessagePort")
```
Comment (bisby) [inline source/utils/misc.brs:102]: My suggestion for this was going to be to set up a message port and observe button presses. So I approve of this step.

**Review by bisby — APPROVED** [review]

**Pull request merged**